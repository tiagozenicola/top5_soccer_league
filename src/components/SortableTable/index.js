import React, { Component } from 'react';
import StyledTable from './style';
import Button from '../Button';


class SortableTable extends Component {
  state = {
    sortProperty: 'goals',
    orientation: -1,
  }

  sort = (field) => {
    const { sortProperty, orientation } = this.state;

    if (field === sortProperty) {
      this.setState({
        sortProperty: field,
        orientation: orientation * -1,
      });
      return;
    }

    this.setState({ sortProperty: field });
  }

  getSortedTeams = () => {
    const { sortProperty, orientation } = this.state;
    const { stats } = this.props;

    if (sortProperty) {
      return stats.sort((a, b) => {
        if (a[sortProperty] < b[sortProperty]) {
          return -1 * orientation;
        }

        if (a[sortProperty] > b[sortProperty]) {
          return 1 * orientation;
        }

        return 0;
      });
    }

    return stats;
  }

  render() {
    const teams = this.getSortedTeams();
    const { numberField } = this.props;

    const listTeams = teams.map(team => (
      <tr key={team.name}>
        <td>{team.name}</td>
        <td>{team[numberField]}</td>
        <td>{team.team}</td>
      </tr>
    ));

    return (
      <div>
        <StyledTable className="soccer_table">
          <thead>
            <tr>
              <th><Button value="Player" name="name" {...this.state} onClick={this.sort} /></th>
              {/* eslint-disable max-len */}
              <th><Button value={numberField} name={numberField} {...this.state} onClick={this.sort} /></th>
              <th><Button value="Team" name="team" {...this.state} onClick={this.sort} /></th>
            </tr>
          </thead>
          <tbody>
            {listTeams}
          </tbody>
        </StyledTable>
      </div>
    );
  }
}


export default (SortableTable);
