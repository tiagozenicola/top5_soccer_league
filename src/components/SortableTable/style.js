import styled from 'styled-components';


const StyledTable = styled.table`
  border: 1px solid red;
  font-size: 12px;
`;

export default StyledTable;
