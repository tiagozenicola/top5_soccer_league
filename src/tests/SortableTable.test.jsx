import React from 'react';
import renderer from 'react-test-renderer';
import enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SortableTable from '../components/SortableTable';
import 'jest-styled-components';

enzyme.configure({ adapter: new Adapter() });

it('renders SortableTable correctly with 1 player', () => {
  const stats = [{ name: 'Eden Hazard', assists: 10, team: 'Chelsea' }];
  const sortableTable = renderer.create(<SortableTable stats={stats} numberField="assists" />).toJSON();
  expect(sortableTable).toMatchSnapshot();
});

it('renders SortableTable correctly with 2 players sort asc', () => {
  const stats = [{ name: 'Eden Hazard', assists: 10, team: 'Chelsea' }, { name: 'Cristiano Ronaldo', assists: 12, team: 'Juventus' }];
  const sortableTable = renderer.create(<SortableTable stats={stats} numberField="assists" />).toJSON();
  expect(sortableTable).toMatchSnapshot();
});

it('renders SortableTable correctly with 1 player sort desc', () => {
  const stats = [{ name: 'Eden Hazard', assists: 10, team: 'Chelsea' }, { name: 'Lionel Messi', assists: 2, team: 'Barcelona' }];
  const sortableTable = renderer.create(<SortableTable stats={stats} numberField="assists" />).toJSON();
  expect(sortableTable).toMatchSnapshot();
});
