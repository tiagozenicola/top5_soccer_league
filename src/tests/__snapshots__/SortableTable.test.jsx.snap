// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SortableTable correctly with 1 player 1`] = `
.c0 {
  border: 1px solid red;
  font-size: 12px;
}

.c1 {
  color: blue;
  border: none;
  background: none;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

<div>
  <table
    className="soccer_table c0"
  >
    <thead>
      <tr>
        <th>
          <button
            className="c1"
            onClick={[Function]}
          >
            Player
            
          </button>
        </th>
        <th>
          <button
            className="c1"
            onClick={[Function]}
          >
            assists
            
          </button>
        </th>
        <th>
          <button
            className="c1"
            onClick={[Function]}
          >
            Team
            
          </button>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Eden Hazard
        </td>
        <td>
          10
        </td>
        <td>
          Chelsea
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders SortableTable correctly with 1 player sort desc 1`] = `
.c0 {
  border: 1px solid red;
  font-size: 12px;
}

.c1 {
  color: blue;
  border: none;
  background: none;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

<div>
  <table
    className="soccer_table c0"
  >
    <thead>
      <tr>
        <th>
          <button
            className="c1"
            onClick={[Function]}
          >
            Player
            
          </button>
        </th>
        <th>
          <button
            className="c1"
            onClick={[Function]}
          >
            assists
            
          </button>
        </th>
        <th>
          <button
            className="c1"
            onClick={[Function]}
          >
            Team
            
          </button>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Eden Hazard
        </td>
        <td>
          10
        </td>
        <td>
          Chelsea
        </td>
      </tr>
      <tr>
        <td>
          Lionel Messi
        </td>
        <td>
          2
        </td>
        <td>
          Barcelona
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders SortableTable correctly with 2 players sort asc 1`] = `
.c0 {
  border: 1px solid red;
  font-size: 12px;
}

.c1 {
  color: blue;
  border: none;
  background: none;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

<div>
  <table
    className="soccer_table c0"
  >
    <thead>
      <tr>
        <th>
          <button
            className="c1"
            onClick={[Function]}
          >
            Player
            
          </button>
        </th>
        <th>
          <button
            className="c1"
            onClick={[Function]}
          >
            assists
            
          </button>
        </th>
        <th>
          <button
            className="c1"
            onClick={[Function]}
          >
            Team
            
          </button>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Eden Hazard
        </td>
        <td>
          10
        </td>
        <td>
          Chelsea
        </td>
      </tr>
      <tr>
        <td>
          Cristiano Ronaldo
        </td>
        <td>
          12
        </td>
        <td>
          Juventus
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
